spec/models/campaign_spec.rb

```diff
@@ -115,18 +116,28 @@
         value: 100_00,
         gift_card_detail: gift_card_detail3
       )
-    end
 
-    it 'returns gift card amounts' do
       expect(campaign.amount_raised).to eq(100_00)
       expect(campaign.last_contribution).to eq(Time.current + 1.day)
     end
   end
+  
+  context 'with amount raised for mega gam campaigns' do
+    let!(:campaign) { create(:campaign, :with_sellers_distributors, :with_project, seller: nil) }
+
+    it 'returns payment intent amounts' do
+      # Expect these 2 to be counted. :with_line_items adds line items of value 600.
+      payment_intent_1 = create(:payment_intent, :with_line_items, campaign: campaign, successful: true)
+      payment_intent_2 = create(:payment_intent, :with_line_items, campaign: campaign, successful: true)
+      # Do not expect this one to be counted since it's not successful.
+      payment_intent_3 = create(:payment_intent, :with_line_items, campaign: campaign, successful: false)
```

Comment:
mind adding these into a subject?
```
  context 'with amount raised for mega gam campaigns' do
    let!(:campaign) { create(:campaign, :with_sellers_distributors, :with_project, seller: nil) }
    subject do 
      # Expect these 2 to be counted. :with_line_items adds line items of value 600.
      payment_intent_1 = create(:payment_intent, :with_line_items, campaign: campaign, successful: true)
      payment_intent_2 = create(:payment_intent, :with_line_items, campaign: campaign, successful: true)
      # Do not expect this one to be counted since it's not successful.
      payment_intent_3 = create(:payment_intent, :with_line_items, campaign: campaign, successful: false)
    end

    it 'returns payment intent amounts' do
      subject 
      expect(campaign.amount_raised).to eq 1200
    end
  end
```

Reply:
or inside of a 
```
before do
  ... 
end
```

Reply:
or inside of a method in this context
```
def create_mega_gam_payment_intents
   ...
end
it 'returns payment intent amounts' do
  create_mega_gam_payment_intents
  ...
```

Reply:
done